src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTest.kt

```diff
@@ -699,21 +700,27 @@ class CargoTestRunnerTest : CargoTestRunnerTestBase() {
         val sourceElement = cargoProjectDirectory.toPsiDirectory(project)!!
         val configuration = createTestRunConfigurationFromContext(PsiLocation.fromPsiElement(sourceElement))
         val root = executeAndGetTestRoot(configuration)
+        val scope = GlobalSearchScope.allScope(project)
 
         val test = root.findTestByName("sandbox::test")
-        assertTrue("cargo:test://sandbox-[a-f0-9]{16}::test".toRegex().matches(test.locationUrl ?: ""))
+        val testDesc = test.getLocation(project, scope)?.openFileDescriptor!!
+        assertEquals(71, testDesc.offset)
```

Comment:
Actually, I don't like that we check raw offsets because it's quite hard to understand if it's correct or not.
I suggest making real navigation and compare actual editor state with expected
Something like
```kotlin
myFixture.configureFromExistingVirtualFile(cargoProjectDirectory.findFileByRelativePath("src/lib.rs")!!)
desc.navigateIn(myFixture.editor)
myFixture.checkResult(expectedText)
```

Of course, it will make this test much bigger but at the same time much clearer
What do you think?
If you don't like a lot of expected code in the test, you can place expected text in test data

Reply:
@Undin Done.